osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ActionMapperService.java

```diff
@@ -228,54 +228,67 @@ private void postConstruct() {
         CLASS_TO_MAPPER_MAP.put(ScanMbusChannelsRequestData.class, this.configurationMapper);
     }
 
-    public BundleMessagesRequestDto mapAllActions(final BundleMessageRequest bundleMessageDataContainer,
+    public BundleMessagesRequestDto mapAllActions(final BundleMessageRequest bundleMessageRequest,
             final SmartMeter smartMeter) throws FunctionalException {
 
         final List<ActionDto> actionValueObjectDtoList = new ArrayList<>();
 
-        for (ActionRequest action : bundleMessageDataContainer.getBundleList()) {
-
-            @SuppressWarnings("unchecked")
-            // suppress else the compiler will complain
-            final CustomValueToDtoConverter<ActionRequest, ActionRequestDto> customValueToDtoConverter = (CustomValueToDtoConverter<ActionRequest, ActionRequestDto>) CUSTOM_CONVERTER_FOR_CLASS
-                    .get(action.getClass());
-
-            if (customValueToDtoConverter != null) {
-                actionValueObjectDtoList.add(new ActionDto(customValueToDtoConverter.convert(action, smartMeter)));
-            } else {
-                final ConfigurableMapper mapper = CLASS_TO_MAPPER_MAP.get(action.getClass());
-                final Class<? extends ActionRequestDto> clazz = CLASS_MAP.get(action.getClass());
-                if (mapper != null) {
-
-                    if (action instanceof MbusActionRequest) {
-                        action = this.verifyAndFindChannelForMbusRequest((MbusActionRequest) action, smartMeter);
-                    }
-
-                    actionValueObjectDtoList.add(new ActionDto(this.performDefaultMapping(action, mapper, clazz)));
-                } else {
-                    throw new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR,
-                            ComponentType.DOMAIN_SMART_METERING,
-                            new AssertionError("No mapper defined for class: " + clazz.getName()));
-                }
-            }
+        for (final ActionRequest action : bundleMessageRequest.getBundleList()) {
+            actionValueObjectDtoList.add(this.mapActionWithMapper(smartMeter, action));
         }
         return new BundleMessagesRequestDto(actionValueObjectDtoList);
     }
 
+    private ActionDto mapActionWithMapper(final SmartMeter smartMeter, final ActionRequest action) throws FunctionalException {
+        @SuppressWarnings("unchecked")
+        // TODO: fix this
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)